tslib/src/caching/redis-service.ts

```diff
@@ -0,0 +1,66 @@
+import Redis, { RedisOptions } from 'ioredis'
+import config from '../config'
+import * as Logger from '../common/ILogger'
+
+export interface IRedisService {
+  /**
+   *
+   * @param {string} key the unique string key of the redis entry to fetch
+   * @returns {Promise<any | null>} an awaitable promise containing the item if it exists, or undefined if it does not
+   */
+  get (key: string): Promise<any | null>
+  /**
+   *
+   * @param {string} key the unique string key of the redis entry to persist
+   * @param {object | string} value the item to persist in redis cache
+   * @param {number} ttl how long to store the item in seconds
+   * @returns {Promise<void} an awaitable promise
+   */
+  setex (key: string, value: string | object, ttl: number): Promise<void>
```

Comment:
Why is this called `setex` rather than `set`?

Reply:
it matches the underlying redis command 'set with an explicit expiration' - https://redis.io/commands/setex

Reply:
Perfect! 😊